storage_controller/src/service.rs

```diff
@@ -6378,8 +6387,8 @@ impl Service {
                 let state = client.step_down(&self.cancel).await;
```

Comment:
Above there is "TODO: jwt token", does that cover cases like jwt is not accepted?

Reply:
> does that cover cases like jwt is not accepted?

So this case falls in the generic case of not being able to receive an answer to the step-down request. In this scenario, we give up and the storage controller proceeds with building the observed state manually and marking itself as the leader in the database. Once the new pod becomes ready, k8s will kill the previous one. This is actually tested in test_storage_controller_leadership_transfer[step_down_times_out=True].

The TODO is about giving the storage controllers in staging/prod the correct JWT token to be able to talk to its peers.

Reply:
Makes sense, thanks.